src/test/java/com/tailf/jnc/framing/FramingTest.java

```diff
@@ -168,14 +171,21 @@ public void twoSeparatedFrames() throws IOException {
         byte[] data = frames.getBytes(StandardCharsets.UTF_8);
         PipedOutputStream po = new PipedOutputStream();
         PipedInputStream pi = new PipedInputStream(po);
-        for (int i = 1; i < frames.length(); i++) {
-            int bytesplit = frames.substring(0, i).getBytes(StandardCharsets.UTF_8).length;
-            DataReader rdr = new TestDataReader(po, data, bytesplit);
-            Framer framer = Framing.END_OF_MESSAGE.newSessionFramer(rdr, pi,
-                                                             nullOutputStream());
-            assertEquals("should restore both frames from parts",
-                         INTL_FRAME + LONGER_FRAME,
-                         framer.parseFrame() + framer.parseFrame());
+        try {
+            for (int i = 1; i < frames.length(); i++) {
+                int bytesplit = frames.substring(0, i).getBytes(StandardCharsets.UTF_8).length;
+                DataReader rdr = new TestDataReader(po, data, bytesplit);
+                Framer framer = Framing.END_OF_MESSAGE.newSessionFramer(rdr, pi,
+                nullOutputStream());
+                assertEquals("should restore both frames from parts",
+                INTL_FRAME + LONGER_FRAME,
+                framer.parseFrame() + framer.parseFrame());
+            }
+        } catch (IOException ex) {
+            throw ex;
+        } finally {
+            pi.close();
```

Comment:
Can you use the resource auto-closing here too? Sorry for not notifying you before.

Reply:
fixed